data/sqlutil/macros.go

```diff
@@ -123,119 +125,96 @@ var DefaultMacros = Macros{
 	"column":     macroColumn,
 }
 
-func trimAll(s []string) []string {
-	r := make([]string, len(s))
-	for i, v := range s {
-		r[i] = strings.TrimSpace(v)
-	}
-
-	return r
+type macroMatch struct {
+	full string
+	args []string
 }
 
 // getMacroMatches extracts macro strings with their respective arguments from the sql input given
 // It manually parses the string to find the closing parenthesis of the macro (because regex has no memory)
-func getMacroMatches(input string, name string) ([][]string, error) {
-	macroName := fmt.Sprintf("\\$__%s\\b", name)
-	matchedMacros := [][]string{}
-
-	rgx, err := regexp.Compile(macroName)
+func getMacroMatches(input string, name string) ([]macroMatch, error) {
+	rgx, err := regexp.Compile(fmt.Sprintf(`\$__%s\b`, name))
 	if err != nil {
 		return nil, err
 	}
 
-	// get all matching macro instances
-	matched := rgx.FindAllStringIndex(input, -1)
-	if matched == nil {
-		return nil, nil
+	var matches []macroMatch
+	for _, window := range rgx.FindAllStringIndex(input, -1) {
+		start, end := window[0], window[1]
+		args, length := parseArgs(input[end:])
+		if length < 0 {
+			return nil, fmt.Errorf("failed to parse macro arguments (missing close bracket?)")
+		}
+		matches = append(matches, macroMatch{full: input[start : end+length], args: args})
+	}
+	return matches, nil
+}
+
+// parseArgs looks for a bracketed argument list at the beginning of
+// argString. If one is present, returns a list of whitespace-trimmed
+// arguments and the length of the string comprising the argument list.
+func parseArgs(argString string) ([]string, int) {
+	if argString == "" || argString[0] != '(' {
+		return nil, 0
 	}
 
-	for matchedIndex := 0; matchedIndex < len(matched); matchedIndex++ {
-		var macroEnd = 0
-		var argStart = 0
-		// quick exit from the loop, when we encounter a closing bracket before an opening one (ie "($__macro)", where we can skip the closing one from the result)
-		var forceBreak = false
-		macroStart := matched[matchedIndex][0]
-		inputCopy := input[macroStart:]
-		cache := make([]rune, 0)
-
-		// find the opening and closing arguments brackets
-		for idx, r := range inputCopy {
-			if len(cache) == 0 && macroEnd > 0 || forceBreak {
-				break
+	var args []string
+	depth := 0
+	arg := []rune{}
+
+	for i, r := range argString {
+		switch r {
+		case '(':
+			depth++
+			if depth == 1 {
+				continue
+			}
+		case ')':
+			depth--
+			if depth == 0 {
+				// this is the closing bracket
+				args = append(args, strings.TrimSpace(string(arg)))
+				return args, i + 1
 			}
-			switch r {
-			case '(':
-				cache = append(cache, r)
-				if argStart == 0 {
-					argStart = idx + 1
-				}
-			case ')':
-				l := len(cache)
-				if l == 0 {
-					macroEnd = 0
-					forceBreak = true
-					break
-				}
-				cache = cache[:l-1]
-				macroEnd = idx + 1
-			default:
+		case ',':
+			if depth == 1 {
+				// top level of the bracketed arg list
+				args = append(args, strings.TrimSpace(string(arg)))
```

Comment:
The current version of `sqlds` only trims the last argument, where the previously migrated version in this repo trims them all. Trimming them all seems more correct.